### PR TITLE
[JENKINS-37816] Upgrade folders plugin dependency to 5.0 and restore API compatability

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-37816](https://issues.jenkins-ci.org/browse/JENKINS-37816)

@reviewbybees

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/uniqueid/impl/FolderIdStore.java [hunk 1]
```diff
@@ -32,16 +35,19 @@ public FolderIdStore() {
 
     @Override
     public void remove(Folder folder) throws IOException {
-        DescribableList<FolderProperty<?>,FolderPropertyDescriptor> properties = folder.getProperties();
-
-        for (Iterator<FolderProperty<?>> itr = properties.iterator(); itr.hasNext(); ) {
-            FolderProperty<?> prop = itr.next();
+        DescribableList<AbstractFolderProperty<?>,AbstractFolderPropertyDescriptor> properties = folder.getProperties();
+        boolean needSave = false;
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/uniqueid/impl/FolderIdStore.java]: needSave was unrelated, but ok

**Comment by jglick** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
🐝

**Comment by rsandell** [comment]
:bee:

**Comment by stephenc** [comment]
@reviewbybees done

**Comment by stephenc** [comment]
As maintainer 👍

**Pull request merged**